client/layout/masterbar/notifications.jsx

```diff
@@ -57,6 +59,9 @@ export default React.createClass( {
 			this.props.onClick( this.state.isShowingPopover );
 
 			if ( this.state.isShowingPopover ) {
+				this.props.recordTracksEvent( 'calypso_notification_open', {
+					unread_notifications: store.get( 'wpnotes_unseen_count' )
+				} );
```

Comment:
This is just a matter of style, but I like to challenge us to think of actions a UI component might inherently take and then pass in parameters which _do_ those things. On that point, analytics are rarely something that a component actually wants to do. It just happens to be that we tag along and track the action. It may not be possible here since we're not controlling any specific action, but here is one way I might consider injecting this here…

```js
this.props.recordOpening( store.get( 'wpnotes_unseen_count' ) );

const mapDispatchToProps = dispatch => ( {
	recordOpening: unread_notifications => dispatch( recordTracksEvent( 'calypso_notification_open', { unread_notifications } ) ),
} );
```